sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGenerateExec.scala

```diff
@@ -53,12 +53,54 @@ class GpuGenerateExecSparkPlanMeta(
   }
 
   override def convertToGpu(): GpuExec = {
-    GpuGenerateExec(
-      childExprs.head.convertToGpu().asInstanceOf[GpuGenerator],
-      gen.requiredChildOutput,
-      gen.outer,
-      gen.generatorOutput,
-      childPlans.head.convertIfNeeded())
+    // if child expression contains Stack, use GpuExpandExec instead
+    if (childExprs.exists(_.isInstanceOf[GpuStackMeta])) {
+      val stackMeta = childExprs.filter(_.isInstanceOf[GpuStackMeta]).head
+      val numRows = stackMeta.childExprs.head.convertToGpu()
+          .asInstanceOf[GpuLiteral].value.asInstanceOf[Int]
+      val numFields = Math.ceil((stackMeta.childExprs.length - 1.0) / numRows).toInt
+      val projections: Seq[Seq[Expression]] = for (row <- 0 until numRows) yield {
+        val childExprsGpu = childExprs.filterNot(_.isInstanceOf[GpuStackMeta])
+            .map(_.convertToGpu())
+        val otherProj: Seq[Expression] = for (req <- childExprsGpu) yield {
+          req
+        }
+        val stackProj: Seq[Expression] = for (col <- 0 until numFields) yield {
+          val index = row * numFields + col
+          if (index >= stackMeta.childExprs.tail.length) {
+            val typeInfo = stackMeta.childExprs.tail(col).dataType
+            GpuLiteral(null, typeInfo)
+          } else {
+            stackMeta.childExprs.tail(index).convertToGpu()
+          }
+        }
+        otherProj ++ stackProj
+      }
```

Comment:
Better to move this into a separate method named e.g. "projectionsForStack". And add comment on why and how to implement GpuStack. Because this is not a normal way we add GPU support for an expression.